webapp/src/main/new/New.tsx

```diff
@@ -121,21 +141,21 @@ export const New = () => {
           />
           <button
             type="submit"
-            onClick={() => {
-              applyPostValidation();
-              if (validationErrors.size === 0) {
-                // FIXME: revise this conditional to use ValidationErrorMap
-                if (isOnline) {
-                  // Submit to cloud when online
-                  insertCensus(data).catch((error) => setErrors(error));
-                } else {
-                  // Save locally when offline
-                  data.forEach((census) =>
-                    userInputStore
-                      ?.setItem(census.CensusId.toString(), census)
-                      .catch((error: any) => console.error(error))
-                  );
-                }
+            onClick={async () => {
+              // FIXME: Post validation is removed for now because it conflicts with cloud validation.
+              // applyPostValidation();
+              if (isOnline) {
+                // Submit to cloud when online
+                await insertCensus(data)
```

Comment:
this looks strange. you already have `.then()`

Reply:
Oh yeah this is leftover from trying out things to make the render happen. That's just a mistake